server/src/main/java/org/apache/druid/rpc/ServiceClientImpl.java

```diff
@@ -200,7 +200,12 @@ public void onSuccess(@Nullable final Either<StringFullResponseHolder, FinalType
                         final String newUri = result.error().getResponse().headers().get("Location");
 
                         if (redirectCount >= MAX_REDIRECTS) {
-                          retVal.setException(new RpcException("Service [%s] issued too many redirects", serviceName));
+                          retVal.setException(new RpcException(
+                              "Service [%s] redirected too many times [%d] to invalid url %s",
```

Comment:
The error message here isn't right: the url isn't known to be invalid at this point. It may very well be valid. It just exceeded the limit on number of redirects.